File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        senitel1, senitel2 = ListNode(), ListNode()
        senitel1.next = list1
        senitel2.next = list2
        r = senitel2
        p, q = list1, list2
        while(p and q):
            if p.val > q.val:
                r.next = q
                r = q
                q = q.next                
            elif p.val <= q.val:
                r.next = p
                r = p
                p = p.next
        if p:
            r.next = p
        if q:
            r.next = q
        return senitel2.next
                

                
        

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dumb = new ListNode(0);
        ListNode p1 = list1;
        ListNode p2 = list2;
        ListNode q = dumb;
        while(p1 != null || p2 != null){
            int p1Val = 101, p2Val = 101;
            if(p1 != null){
                p1Val = p1.val;
            }
            if(p2 != null){
                p2Val = p2.val;
            }
            ListNode add = new ListNode(Math.min(p1Val, p2Val));
            q.next = add;
            q = q.next;
            if(p1Val == add.val){
                p1 = p1.next;
            }else{
                p2 = p2.next;
            }
            
        }
        return dumb.next;
        
        
        
    }
}